src/marker-layer.coffee

```diff
@@ -301,7 +308,7 @@ class MarkerLayer
     ranges = @index.dump()
     for id in Object.keys(@markersById)
       marker = @markersById[id]
-      result[id] = marker.getSnapshot(Range.fromObject(ranges[id]), false)
+      result[id] = marker.getSnapshot(Range.fromObject(ranges[id]))
```

Comment:
Should we be passing our `@maintainHistory` flag as the second argument to `getSnapshot` here, so that we only include the marker if we need it?

Reply:
Nice idea.

Reply:
Actually, the only call site is [already conditional](https://github.com/atom/text-buffer/blob/fix-marker-resurrection/src/text-buffer.coffee#L1957) on `maintainHistory` being true.

Reply:
Oh yeah, of course.